lib/licensed/sources/gradle.rb

```diff
@@ -122,49 +138,51 @@ def configurations
         end
       end
 
-      def self.add_gradle_license_report_plugins_block(gradle_build_file)
-
-        if gradle_build_file.include? "plugins"
-          gradle_build_file.gsub(/(?<=plugins)\s+{/, " { id 'com.github.jk1.dependency-license-report' version '1.16'")
-        else
-
-          gradle_build_file = " plugins { id 'com.github.jk1.dependency-license-report' version '1.16' }" + gradle_build_file
-        end
+      def project
+        @project ||= config.dig("gradle", "project")
       end
 
-      def self.gradle_command(*args, path:, executable:, configurations:)
-        gradle_build_file = File.read("build.gradle")
-
-        if !gradle_build_file.include? "dependency-license-report"
-          gradle_build_file = Licensed::Sources::Gradle.add_gradle_license_report_plugins_block(gradle_build_file)
-        end
-
+      def self.gradle_command(*args, path:, executable:, configurations:, project:)
```

Comment:
it looks like you're getting rid of the `add_grade_license_report_plugins_block` function, as well as changing the name and contents of the created file.

Would you mind adding some context behind the purpose of the changes and any specifics about the implementation?  

Also, it looks like you're using some new properties in the init script, do you know how far back that new syntax is supported?  Licensed generally aims to support all LTS versions of tools/frameworks so I want to make sure this change won't cause any backwards compatibility failures for users on older supported versions of gradle.

Reply:
https://github.com/github/licensed/pull/583#discussion_r1041383560